### PR TITLE
[Paddle] Add RMSNorm, RoPE and SwiGLU

### PR DESCRIPTION
**Pull request by Wong4j**
Please merge PR#595 before this one.
1. Add normalization option (`LayerNorm or RMSNorm`) in `LayerNormLinear`, `LayerNormMLP`, `MultiHeadAttention` and `TransformerLayer`.
2. Add Rotary Position Embedding.
3. Add SwiGLU.

### CODE REVIEW

File: tests/paddle/test_operators.py [hunk 1]
```diff
@@ -296,6 +294,55 @@ def test_gelu_bwd_fp8(fp8_dtype):
         assert_allclose(x_grad_t, x.grad.T, rtol=0.1, atol=0.01)
         assert_allclose(dbias, x.grad.sum(axis=0), rtol=0.1, atol=0.01)
 
+    @staticmethod
+    def test_silu_bf16():
+        """
+        Test BF16 SiLU Forward
+        """
```
Comment (timmoon10) [inline tests/paddle/test_operators.py:301]: SwiGLU and SiLU/swish are different activations:
```suggestion
    def test_swiglu_bf16():
        """
        Test BF16 SwiGLU Forward
        """
```
There are a few other tests that are called "silu" but use SwiGLU.

**Review by timmoon10 — CHANGES_REQUESTED** [review]
Mostly looks good, although there are a few minor issues.

File: tests/paddle/test_operators.py (hunk 1 above)
Comment (Wong4j) [reply tests/paddle/test_operators.py]: Done

**Comment by jeng1220** [comment]
LGTM

**Review by timmoon10 — APPROVED** [review]
LGTM

**Pull request merged**